src/objective/multiclass_objective.hpp

```diff
@@ -46,11 +60,12 @@ class MulticlassLogloss: public ObjectiveFunction {
           auto p = rec[k];
           size_t idx = static_cast<size_t>(num_data_) * k + i;
           if (label_int_[i] == k) {
-            gradients[idx] = static_cast<score_t>(p - 1.0f);
+            gradients[idx] = static_cast<score_t>(p - 1.0f) * label_pos_weights_[k];
+            hessians[idx] = static_cast<score_t>(2.0f * p * (1.0f - p))* label_pos_weights_[k];
           } else {
             gradients[idx] = static_cast<score_t>(p);
+            hessians[idx] = static_cast<score_t>(2.0f * p * (1.0f - p));
```

Comment:
I would expect gradients and hessians to be multiplied by the weight corresponding to their class. Which is the reason behind only doing so for the positive samples in this binary classifier?

I would like to dig deeper into the loss itself. Is this the gradient from the loss function at https://github.com/Microsoft/LightGBM/blob/b884da85f1234e12b77f09ce054b6c43b0329454/src/metric/multiclass_metric.hpp ?